### ISSUE TITLE
Segmentation Fault when loading SofaPython3.dll

### ISSUE THREAD

**Issue by lytyo** [sofa-framework/sofa#1672]
Hello,

I try to use SofaPython3 with sofa (master branch) on windows. I use MSVC compiler.

I did the following :

1. Installed pybind11 via pip3.9.
2. set CMake variable `pybind11_DIR` = `C:/Program Files/Python39/Lib/site-packages/pybind11/share/cmake/pybind11`.
3. set CMake  variable`SOFA_FETCH_PYTHON3` to true.
4. set CMake variables `PLUGIN_SOFAPYTHON3`, `PLUGIN_SOFAPYTHON3_BINDINGS`, `PLUGIN_SOFAPYTHON3_DOCS` and `PLUGIN_SOFAPYTHON3_EXAMPLES` to true.
4. 'Configure' then 'Generate' using CMake's GUI.
5. Build with MSVC -> success (release mode).

I encounter a segfault error when loading SofaPython3.dll via the Sofa GUI : Edit>Plugin Manager...>Add...>SofaPython3.dll.
--> **exception Microsoft C++ : pybind11::error_already_set**

Do you have any idea what is causing this error ?

Thanks

**Comment by lytyo** [comment]
I managed to solve the issue. The main reason appeared to be the location of the "python3" directory. Apperently, it should be located one directory higher than the directory containing all the built DLLs. 
So I changed the the CMake variable `SP3_PYTHON_PACKAGES_DIRECTORY` from "python3/site-packages" to "**bin**/python3/site-packages".
I also downgraded python from 3.9.1 to 3.7.9 and pulled the last commits from sofa master but I think the issue came from the python3 directory.

**Comment by fredroy** [comment]
Hi @lytyo , sorry for the delay.
Do you still have problem with the v20.12 release of SOFA ? SofaPython3 is being shipped inside and emphasis has been placed on compilation/build/install with python path, etc.
Thanks !

**Comment by lytyo** [comment]
Hi @fredroy, thank you for your reply,

I recently pulled Sofa 20.12 and SofaPython3 and I did not observe this issue. But it is maybe because my cmake variable `SP3_PYTHON_PACKAGES_DIRECTORY` was already set to `bin/python3/site-packages`.

**Comment by lytyo** [comment]
Hi @fredroy,

I have an update concerning this issue. I tried compiling SOFA with SofaPython3 in Debug mode and I encountered the issue again.
Firstly, in Visual Studio, I had to add the Python3 libs folder as additional library reference to the python3 related projects. (Otherwise python39.lib was not found)

In `PythonEnvironment::Init()`, the line `py::module::import("SofaRuntime");` causes the crash. So apparently the import fails. 

I think the cause could be that the .pyd files make reference to unreachable dependencies. So I ran Dependencies and did not notice anything wrong :
![image](https://user-images.githubusercontent.com/24836743/108839575-87195400-75d5-11eb-8438-1cbf28fe8bb2.png)

I precise that I installed the Python Debug binairies (so python39_d.dll exists).
